### ISSUE TITLE
fromBitmap very slow

### ISSUE THREAD

**Issue by dericdesta** [jcodec/jcodec#50]
Hi everyone,
I have got several images which I want to encode to video. I scale down the bitmaps using inSampleSize:

```
                SequenceEncoder encoder = new SequenceEncoder(file);

                for (int i = 0; i < encode_images.size(); i++) {
                    // First decode with inJustDecodeBounds=true to check
                    // dimensions
                    Log.i("add_frame", encode_images.get(i));
                    final BitmapFactory.Options options = new BitmapFactory.Options();
                    options.inJustDecodeBounds = true;
                    BitmapFactory.decodeFile(encode_images.get(i), options);

                    // Calculate inSampleSize
                    options.inSampleSize = calculateInSampleSize(options, 180,
                            240);

                    // Decode bitmap with inSampleSize set
                    options.inJustDecodeBounds = false;

                    Picture pic = fromBitmap(BitmapFactory.decodeFile(
                            encode_images.get(i), options));
                    encoder.encodeNativeFrame(pic);

                }
                encoder.finish();
```

As you can see, my bitmaps are not that large. Nevertheless "fromBitmap(...)" takes much time (~1-2 minutes). Does anyone have a solution for that?

Thanks.

**Comment by sangramshitole** [comment]
Hello,
Any update or workaround for this issue? Thanks!

**Comment by svitvitskiy** [comment]
TLDR; The speed will never be comparable to the hardware encoder or your PC.

The entire encoder is implemented in pure Java and this will be slow on a mobile phone. First, mobile phones have low power (low performance) CPU. Second, java implementation in Android could be better.

The workarounds:
  - Support complexity settings in encoder (motion search range etc). Will look into this eventually.
  - Implement the computationally intensive parts in [RenderScript](https://developer.android.com/guide/topics/renderscript/compute.html). Will do this eventually.
